blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BlueTable.java

```diff
@@ -0,0 +1,32 @@
+package io.jenkins.blueocean.rest.model;
+
+import com.google.common.annotations.Beta;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A Table-like data structure for serving tabular data
+ */
+@ExportedBean
+@Beta
+public abstract class BlueTable {
+
+    public static final String COLUMNS = "columns";
+    public static final String ROWS = "rows";
+
+    @Exported(name = COLUMNS, inline = true)
+    public abstract Map<String, String> getColumns();
+
+    @Exported(name = ROWS)
+    public abstract List<Row> getRows();
```

Comment:
How about `getRows()` and `getColumns()` sizes? Wonder if it could be expensive, paginate these? `PagedResponse` won't work for these, its for individual collection resource. What we need here is paginated collection inside a json object. 

```json
{
    "start":0,
    "limit":100,
    "rows":[{},{}]
}
```